src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,305 @@
+    // _windowSizes and _minWindowCount are int64 so that we can have the full positive range of the uint32 it comes
+    // in as but can still support negative numbers.
+    Components::CircularBuffer<InputT>              _buffer;
+    const std::uint32_t                             _windowSize;
+    const std::uint32_t                             _horizon;
+    const std::uint32_t                             _minWindowSize;
+    const AnalyticalRollingWindowCalculation        _windowCalculation;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        typename BaseType::TransformedType         results(_horizon);
+        
+        _buffer.push(input);
+        size_t bufferSize = _buffer.size();
+
+        double result;
+        size_t numElements;
+        size_t startingOffset;
+
+        for (std::uint32_t offset = 0; offset < _horizon; ++offset){
+
+            // If we don't have enough elements then output NaN
+            if ((_horizon - offset) + _minWindowSize > bufferSize) {
+                result = Traits<double>::CreateNullValue();
+
+            // If we have strictly less elements then the window size, but more then the minimum size
+            } else {
+                if (bufferSize - (_horizon - offset) < _windowSize) {
+                    numElements = bufferSize - (_horizon - offset); //asfd
+                    startingOffset = 0;
+                } else {
+                    numElements = _windowSize;
+                    startingOffset = offset - (_buffer.capacity() - bufferSize);
+                }
+
+                const std::tuple<Components::CircularBuffer<InputT>::iterator, Components::CircularBuffer<InputT>::iterator> range = _buffer.range(numElements, startingOffset);
+                const Components::CircularBuffer<InputT>::iterator start_iter = std::get<0>(range);
+                const Components::CircularBuffer<InputT>::iterator end_iter = std::get<1>(range);
+
+                result = Calculators::MeanCalculator<BaseType::InputType>::execute(start_iter, end_iter);
+
+            }
+
+            results[offset] = result;
+        }
+
+        callback(results);
+    }
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         AnalyticalRollingWindowEstimator
+///  \brief         Estimator that creates `AnalyticalRollingWindowTransformer`.
+///
+template <
+    typename InputT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class AnalyticalRollingWindowEstimator :
+    public TransformerEstimator<InputT, std::vector<double>> {

```

Comment:
>TransformerEstimator<InputT, std::vector<double>> [](start = 11, length = 49)

You can base this class on `Components::InferenceOnlyEstimatorImpl` will simplify the implementation.

Reply:
I had an issue when doing that. Let me try again and see.

Reply:
I may not be doing the override correctly, but when I do this it has issues because the transformer doesn't have a default constructor.